PokemonAPI/PokemonAPI.BusinessLayer/Implementations/DomainServices/TrainerService.cs

```diff
@@ -0,0 +1,36 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using PokemonAPI.BusinessLayer.Interfaces;
+using PokemonAPI.DomainLayer.Entities;
+using PokemonAPI.PersistenceLayer.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace PokemonAPI.BusinessLayer.Implementations.DomainServices
+{
+    internal class TrainerService : ITrainerService
+    {
+        private readonly IRepository<Trainer> _repository;
+        private readonly IMapper _mapper;
+
+        public TrainerService(IRepository<Trainer> repository, IMapper mapper)
+        {
+            _repository = repository;
+            _mapper = mapper;
+        }
+
+        public async Task<IEnumerable<TrainerDTO>> GetAll(CancellationToken cancellationToken = default)
+        {
+            var trainers = await _repository.GetAll(
+                include: source => source
+                    .Include(t => t.CaughtPokemons),
```

Comment:
just to make things clear
This is a lambda expression

Here, with parameter **include** you are defining which entities you wish to **eagerly load**
Entity Framework won't load our navigation properties unless we tell it to.

https://docs.microsoft.com/en-us/ef/core/querying/related-data
You can either:
Eagerly load
Implicitly load
Lazy load

Eager loading means that the related data is loaded from the database as part of the initial query.

Lazy loading means that the related data is transparently loaded from the database when the navigation property is accessed.

To enable lazy loading we need to configure some things for the Entity framework in our project. But we don't need it.

Read more in the official documentation
https://docs.microsoft.com/en-us/ef/core/querying/related-data